source/Calamari/Kubernetes/AwsAuthConventionFactoryFactory.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using Calamari.Common.Plumbing.Logging;
+using Calamari.Common.Plumbing.Variables;
+using Calamari.Deployment.Conventions;
+
+namespace Calamari.Kubernetes
+{
+    public class AwsAuthConventionFactoryFactory
```

Comment:
You read that right AwsAuthConventionFactory... Factory!

Reply:
If you inject a Lazy, it does check if the type in the Lazy is registered, but it doesn't check that type's dependencies so if you create another Lazy with in that, autofac doesn't complain if it's not actually a registered interface.

Reply:
This feels so dirty 😁 But if it works, it works!
A comment explaining what this does, and why you've done this would be great - I.e. something like explain that we're stopping Autofac from checking the dependency, and that we're doing it because of dotnet 4 causing issues for xyz etc.

Reply:
I wonder if there's another name we could give this which might be less cryptic than FactoryFactory, like AwsAuthConventionFactoryWrapper or something?

Reply:
I have changed the class name and added a long winded comment. I also slightly updated the factory so that it doesn't need to be passed the `ILog` and `IVariables` as I realised they can just be injected.